File: Nian_Karl/SoftwareConstruction/src/org/uva/ql/ast/value/Value.java

package org.uva.ql.ast.value;



public abstract class Value {
	
	private static final UndefinedValue UNDEFINED = new UndefinedValue(); 
	
	public abstract Object getValue();
	public abstract String toString();
	
	public boolean isUndefined(){return true;}
	public Value not() {return UNDEFINED;}
	public Value positive() {return UNDEFINED;}
	public Value negative() {return UNDEFINED;}
	
	public Value plus(Value arg) {return UNDEFINED;}
	public Value minus(Value arg) {return UNDEFINED;}
	public Value multiply(Value arg) {return UNDEFINED;}
	public Value divide(Value arg) {return UNDEFINED;}
	public Value and(Value arg) {return UNDEFINED;}
	public Value or(Value arg) {return UNDEFINED;}
	public Value equal(Value arg) {return UNDEFINED;}
	public Value notEqual(Value arg) {return UNDEFINED;}
	public Value greater(Value arg) {return UNDEFINED;}
	public Value greaterEqual(Value arg) {return UNDEFINED;}
	public Value less(Value arg) {return UNDEFINED;}
	public Value lessEqual(Value arg) {return UNDEFINED;}
	
	public Value intPlus(IntValue arg) {return UNDEFINED;}
	public Value intMinus(IntValue arg) {return UNDEFINED;}
	public Value intMultiply(IntValue arg) {return UNDEFINED;}
	public Value intDivide(IntValue arg) {return UNDEFINED;}
	public Value boolAnd(BoolValue arg) {return UNDEFINED;}
	public Value boolOr(BoolValue arg) {return UNDEFINED;}
	public Value intGreater(IntValue arg) {return UNDEFINED;}
	public Value intGreaterEqual(IntValue arg) {return UNDEFINED;}
	public Value intLess(IntValue arg) {return UNDEFINED;}
	public Value intLessEqual(IntValue arg) {return UNDEFINED;}
	public Value boolEqual(BoolValue arg) {return UNDEFINED;}
	public Value boolNotEqual(BoolValue arg) {return UNDEFINED;}
	public Value intEqual(IntValue arg) {return UNDEFINED;}
	public Value intNotEqual(IntValue arg) {return UNDEFINED;}
	public Value strEqual(StrValue arg) {return UNDEFINED;}
	public Value strNotEqual(StrValue arg) {return UNDEFINED;}
	public Value StrPlus(StrValue arg) {return UNDEFINED;}
}

File: FelipezConde/QuestionnaireLanguage/Evaluation/Values/Value.cs
using System;

namespace Evaluation.Values
{
    public abstract class Value
    {
        public abstract T Accept<T>(IValueVisitor<T> visitor);
        public abstract Value Equal(Value value);
        public abstract Value NotEqual(Value value);
        public virtual Bool Negate() { throw new NotImplementedException(); }
        public virtual Value And(Value value) { throw new NotImplementedException(); }
        public virtual Value Or(Value value) { throw new NotImplementedException(); }
        public virtual Value Greater(Value value) { throw new NotImplementedException(); }
        public virtual Value GreaterEqual(Value value) { throw new NotImplementedException(); }
        public virtual Value Less(Value value) { throw new NotImplementedException(); }
        public virtual Value LessEqual(Value value) { throw new NotImplementedException(); }
        public virtual Value Add(Value value) { throw new NotImplementedException(); }
        public virtual Value Substract(Value value) { throw new NotImplementedException(); }
        public virtual Value Divide(Value value) { throw new NotImplementedException(); }
        public virtual Value Multiply(Value value) { throw new NotImplementedException(); }
        public virtual Value BoolAnd(Bool boolValue) { throw new NotImplementedException(); }
        public virtual Value BoolOr(Bool boolValue) { throw new NotImplementedException(); }
        public virtual Value BoolEqual(Bool boolValue) { throw new NotImplementedException(); }
        public virtual Value BoolNotEqual(Bool boolValue) { throw new NotImplementedException(); }
        public virtual Value StringEqual(Values.String stringValue) { throw new NotImplementedException(); }
        public virtual Value StringNotEqual(Values.String stringValue) { throw new NotImplementedException(); }
        public virtual Value IntegerAdd(Values.Int intValue) { throw new NotImplementedException(); }
        public virtual Value IntegerSubstract(Values.Int intValue) { throw new NotImplementedException(); }
        public virtual Value IntegerDivide(Values.Int intValue) { throw new NotImplementedException(); }
        public virtual Value IntegerMultiply(Values.Int intValue) { throw new NotImplementedException(); }
        public virtual Value IntegerEqual(Values.Int intValue) { throw new NotImplementedException(); }
        public virtual Value IntegerNotEqual(Values.Int intValue) { throw new NotImplementedException(); }
        public virtual Value IntegerGreater(Values.Int intValue) { throw new NotImplementedException(); }
        public virtual Value IntegerGreaterEqual(Values.Int intValue) { throw new NotImplementedException(); }
        public virtual Value IntegerLess(Values.Int intValue) { throw new NotImplementedException(); }
        public virtual Value IntegerLessEqual(Values.Int intValue) { throw new NotImplementedException(); }
    }
}
